packages/ia-topnav/src/primary-nav.js

```diff
@@ -79,9 +79,24 @@ class PrimaryNav extends TrackedElement {
     );
   }
 
+  get isRTL() {
+    // https://stackoverflow.com/questions/12006095/javascript-how-to-check-if-character-is-rtl
+    const ltrChars = 'A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02B8\u0300-\u0590\u0800-\u1FFF'+'\u2C00-\uFB1C\uFDFE-\uFE6F\uFEFD-\uFFFF';
+    const rtlChars = '\u0591-\u07FF\uFB1D-\uFDFD\uFE70-\uFEFC';
+    const rtlCheck = new RegExp('^[^'+ltrChars+']*['+rtlChars+']');
+
+    return rtlCheck.test(this.screenName);
+  }
```

Comment:
Just noting that browsers can also be made to perform this RTL character check natively, by adding a `dir="auto"` attribute to the enclosing element. The first strongly-directional character found in the element's text determines its overall directionality, placing nondirectional characters (like the ellipsis) on the correct side automatically: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir

(Maybe that wouldn't apply to this use case, but just figured I'd point it out as an option to explore)

Reply:
Thanks Laton, yes, aware of `dir="auto"` and that, if I'm not mistaken, requires an rendered element to work on.

Reply:
Yeah, it's true that the RTL checks would take place at render time rather than during JS execution, and the result would not be readily available to our code.

My thinking was that if we can get the browser to do all the RTL detection & rendering work for us natively, then we may not _need_ JS-based RTL checks in the first place. Using `dir="auto"`, the browser can determine which side to visually render the ellipsis on without our JS having to know anything about the directionality of the text.

Reply:
Thank you for your considered discussion. 
I checked with Jim Shelton about having truncation/ellipsis by `max-width` and not 9 characters. Gets rid of a good deal of code, which makes sense to me.